python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py

```diff
@@ -36,17 +41,72 @@ def build_looker_explore_specs(project_dir: Path) -> Sequence[AssetSpec]:
-            upstream_table = view.get("sql_table_name") or view["name"]
-            upstream_table_asset_key = AssetKey(upstream_table.replace("`", "").split("."))
+            upstream_tables: Sequence[exp.Table] = [to_table(view["name"], dialect=sql_dialect)]
+
+            # https://cloud.google.com/looker/docs/derived-tables
+            derived_table_sql = view.get("derived_table", {}).get("sql")
+
+            if derived_table_sql and "$" in derived_table_sql:
+                logger.warn(
+                    f"Failed to optimize derived table SQL for view `{view['name']}` in file {view_path.name}."
+                    " The upstream dependencies for the view will be omitted."
+                )
+
+                upstream_tables = []
+            elif (
+                derived_table_sql
+                # We need to handle the Looker substitution operation ($) properly since the lkml
+                # compatible SQL may not be parsable yet by sqlglot.
+                #
+                # https://cloud.google.com/looker/docs/sql-and-referring-to-lookml#substitution_operator_
+                and "$" not in derived_table_sql
+            ):
+                try:
+                    optimized_derived_table_ast = optimize(
+                        parse_one(sql=derived_table_sql, dialect=sql_dialect),
+                        dialect=sql_dialect,
+                        validate_qualify_columns=False,
+                    )
+
+                    root_scope = build_scope(optimized_derived_table_ast)
+
+                    if not root_scope:
+                        raise Exception("This is bad!")
+
+                    selected_tables = [
+                        source
+                        for scope in cast(Iterator[Scope], root_scope.traverse())
+                        for (_, source) in scope.selected_sources.values()
+                        if isinstance(source, exp.Table)
+                    ]
+
+                    if selected_tables:
+                        upstream_tables = selected_tables
+                except Exception as e:
+                    logger.warn(
+                        f"Failed to optimize derived table SQL for view `{view['name']}` in file {view_path.name}."
+                        " The upstream dependencies for the view will be omitted.\n\n"
+                        f"Exception: {e}"
+                    )
+
+                    upstream_tables = []
+
+            # https://cloud.google.com/looker/docs/reference/param-view-sql-table-name
+            elif sql_table_name := view.get("sql_table_name"):
+                upstream_tables = [to_table(sql_table_name.replace("`", ""), dialect=sql_dialect)]
 
             looker_view_specs.append(
                 AssetSpec(
                     key=AssetKey(["view", view["name"]]),
-                    deps={upstream_table_asset_key},
+                    deps={
+                        AssetKey([part.name.replace("*", "_star") for part in table.parts])
```

Comment:
Any other special chars we should worry about?

Reply:
This was specific to bigquery, but in general, am following https://github.com/dagster-io/dagster/blob/7d0cf6ddb70139155f5ab9ddca59a4adca055183/python_modules/libraries/dagster-dbt/dagster_dbt/utils.py#L41.

- I didn't bother including (`.` -> `_`) since I assume the table name parts can't include that anyways.
- We support hyphens in `AssetKey`, so I didn't include (`-` -> `_`).